### PR TITLE
feat: contract data route

### PR DESCRIPTION
**Pull request by asiaziola**

### CODE REVIEW

File: src/gateway/router/gatewayRouter.ts [hunk 1]
```diff
@@ -1,27 +1,29 @@
...
 
 // get
-gatewayRouter.get('/contracts', contractsRoute);
-gatewayRouter.get('/contract', contractWithSourceRoute);
-gatewayRouter.get('/contracts/:id', contractRoute);
-gatewayRouter.get('/contracts-safe', safeContractsRoute);
-gatewayRouter.get('/search/:phrase', searchRoute);
+gatewayRouter.get("/contracts", contractsRoute);
+gatewayRouter.get("/contract", contractWithSourceRoute);
+gatewayRouter.get("/contract-data/:id", contractDataRoute);
```
Comment (ppedziwiatr) [inline src/gateway/router/gatewayRouter.ts:23]: note to myself - configure cache in cloudfront for this endpoint.

File: src/gateway/router/routes/contractDataRoute.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  if (!isTxIdValid(id as string)) {
+    logger.error('Incorrect contract transaction id.');
+    ctx.status = 500;
+    ctx.body = { message: 'Incorrect contract transaction id.' };
+    return;
+  }
+
+  try {
+    const benchmark = Benchmark.measure();
+    logger.debug('Id', id);
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/contractDataRoute.ts:23]: either remove or at least change to `contractDataRoute Id`

File: src/gateway/router/routes/contractDataRoute.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    }
+  } catch (e: any) {
+    logger.error(e);
+    ctx.status = 500;
+    ctx.body = { message: e };
+  }
+}
+
+async function getContractData(arweave: Arweave, logger: RedStoneLogger, id: string) {
+  const data = await fetch(`https://arweave.net/${id}`)
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/contractDataRoute.ts:51]: just being curious - will it work `this.arweave.transactions.getData`? If so, we should probably use `ArweaveWrapper.txData` here (ArweaveWrapper should be available in the koa context)

File: src/gateway/router/routes/contractDataRoute.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    .then((res) => {
+      return res.arrayBuffer();
+    })
+    .then((data) => {
+      return data;
+    });
+
+  // decompress and decode contract transction data
+  const gunzipPromisified = util.promisify(gunzip);
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/contractDataRoute.ts:60]: can this be moved to a top level?

File: src/gateway/router/routes/contractDataRoute.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  const gunzippedData = await gunzipPromisified(data);
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/contractDataRoute.ts:61]: unzippedData...

File: src/gateway/router/routes/contractDataRoute.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  logger.debug(`Gunzipped data for bundled contract: ${id}`, gunzippedData);
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/contractDataRoute.ts:62]: please remove.

File: src/gateway/router/routes/contractDataRoute.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  const strData = arweave.utils.bufferToString(gunzippedData);
+  logger.debug(`Parsed data for bundled contract: ${id}`, strData);
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/contractDataRoute.ts:64]: please remove.

File: src/gateway/router/routes/contractDataRoute.ts [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  const tx = new Transaction({ ...JSON.parse(strData) });
+  const bufferFromTxData = Buffer.from(tx.data);
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/contractDataRoute.ts:66]: txData

File: src/gateway/router/routes/contractDataRoute.ts [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+  // get contract transaction content type from its tag
+  const contentType = await getContentType(tx);
+  logger.debug(`Content type for id: ${id}: `, contentType);
+
+  return { data: bufferFromTxData, contentType };
+}
+
+async function getContentType(tx: Transaction) {
+  const tagContentType = await tx
```
Comment (ppedziwiatr) [inline src/gateway/router/routes/contractDataRoute.ts:76]: why await? `get` is not async? https://github.com/ArweaveTeam/arweave-js/blob/master/src/common/lib/transaction.ts#L15

File: src/gateway/router/routes/contractDataRoute.ts (hunk 6 above)
Comment (asiaziola) [reply src/gateway/router/routes/contractDataRoute.ts:62]: i don't really get why you want to remove all the debugs. they will just show up if you will switch the logger level to debug and hit the endpoint. i believe if you want to debug this route you would most certainly need them.

File: src/gateway/router/gatewayRouter.ts [hunk 10]
```diff
@@ -1,27 +1,29 @@
...
-import { contractWithSourceRoute } from './routes/contractWithSourceRoute';
-import { interactionRoute } from './routes/interactionRoute';
-import { safeContractsRoute } from './routes/safeContractsRoute';
-import { sequencerRoute } from './routes/sequencerRoute';
-import { interactionsStreamRoute } from './routes/interactionsStreamRoute';
-import { deployContractRoute } from './routes/deployContractRoute';
-import { arweaveInfoRoute } from './routes/arweaveInfoRoute';
-import { interactionsSortKeyRoute } from './routes/interactionsSortKeyRoute';
+import Router from "@koa/router";
+import {contractsRoute} from "./routes/contractsRoute";
```
[the remark this answers is not in the document]
Comment (asiaziola) [reply src/gateway/router/gatewayRouter.ts:2]: -.-

**Pull request merged**